### ISSUE TITLE
[Bug]: SSO is looking for a older auth json

### ISSUE THREAD

**Issue by RafaelClovr** [hashicorp/terraform#33431]
### Terraform Core Version

1.5.1

### AWS Provider Version

5.5.0

### Affected Resource(s)

S3 backend configuration.

### Expected Behavior

I must be able to use properly my SSO profile for the S3 backend when I renew the token with:
`aws sso login --profile testsso`

### Actual Behavior

Actually, I got that error 
```
│ Error: error configuring S3 Backend: no valid credential sources for S3 Backend found.
│ 
│ Please see https://www.terraform.io/docs/language/settings/backends/s3.html
│ for more information about providing credentials.
│ 
│ Error: SSOProviderInvalidToken: the SSO session has expired or is invalid
│
```

But `aws s3 ls --profile testsso` is working.

I had to copy the new json file located in `.aws/sso/cache` as the old file to do a terraform plan succesfully.

### Relevant Error/Panic Output Snippet

```shell
╷
│ Error: error configuring S3 Backend: no valid credential sources for S3 Backend found.
│ 
│ Please see https://www.terraform.io/docs/language/settings/backends/s3.html
│ for more information about providing credentials.
│ 
│ Error: SSOProviderInvalidToken: the SSO session has expired or is invalid
│ 
│ 
│
```

### Terraform Configuration Files

```hcl
terraform {
  required_providers {
    aws = {
      source  = "hashicorp/aws"
      version = "~> 5.4.0"
    }
    cloudflare = {
      source  = "cloudflare/cloudflare"
      version = "~> 4.8.0"
    }
  }
  backend "s3" {
    bucket               = "demo-tf-state"
    key                  = "state"
    profile              = "mysso"
    workspace_key_prefix = "workspaces"
    region               = "eu-west-1"
  }
}

provider "aws" {
  profile = "mysso"
  region  = "eu-west-1"
}
```
And 

```hcl
resource "aws_s3_bucket" "testtf-backend" {
  bucket = "test-tf-state"

  tags = {
    Environment = "production"
  }
}
```

### Steps to Reproduce

1. Configure AWS SSO.
2. Create a s3 bucket
3. Try to use that s3 backend to store the state.

### Debug Output

```
2023-06-26T16:49:37.852+0200 [INFO]  Terraform version: 1.5.1
2023-06-26T16:49:37.852+0200 [DEBUG] using github.com/hashicorp/go-tfe v1.26.0
2023-06-26T16:49:37.852+0200 [DEBUG] using github.com/hashicorp/hcl/v2 v2.16.2
2023-06-26T16:49:37.852+0200 [DEBUG] using github.com/hashicorp/terraform-svchost v0.1.0
2023-06-26T16:49:37.852+0200 [DEBUG] using github.com/zclconf/go-cty v1.12.2
2023-06-26T16:49:37.852+0200 [INFO]  Go runtime version: go1.20
2023-06-26T16:49:37.852+0200 [INFO]  CLI args: []string{"terraform", "plan"}
2023-06-26T16:49:37.852+0200 [TRACE] Stdout is a terminal of width 162
2023-06-26T16:49:37.852+0200 [TRACE] Stderr is a terminal of width 162
2023-06-26T16:49:37.852+0200 [TRACE] Stdin is a terminal
2023-06-26T16:49:37.852+0200 [DEBUG] Attempting to open CLI config file: /home/test/.terraformrc
2023-06-26T16:49:37.852+0200 [DEBUG] File doesn't exist, but doesn't need to. Ignoring.
2023-06-26T16:49:37.852+0200 [DEBUG] ignoring non-existing provider search directory terraform.d/plugins
2023-06-26T16:49:37.852+0200 [DEBUG] ignoring non-existing provider search directory /home/test/.terraform.d/plugins
2023-06-26T16:49:37.852+0200 [DEBUG] ignoring non-existing provider search directory /home/test/.local/share/terraform/plugins
2023-06-26T16:49:37.852+0200 [DEBUG] ignoring non-existing provider search directory /usr/share/ubuntu/terraform/plugins
2023-06-26T16:49:37.852+0200 [DEBUG] ignoring non-existing provider search directory /usr/share/gnome/terraform/plugins
2023-06-26T16:49:37.852+0200 [DEBUG] ignoring non-existing provider search directory /home/test/.local/share/flatpak/exports/share/terraform/plugins
2023-06-26T16:49:37.852+0200 [DEBUG] ignoring non-existing provider search directory /var/lib/flatpak/exports/share/terraform/plugins
2023-06-26T16:49:37.852+0200 [DEBUG] ignoring non-existing provider search directory /usr/local/share/terraform/plugins
2023-06-26T16:49:37.852+0200 [DEBUG] ignoring non-existing provider search directory /usr/share/terraform/plugins
2023-06-26T16:49:37.852+0200 [DEBUG] ignoring non-existing provider search directory /var/lib/snapd/desktop/terraform/plugins
2023-06-26T16:49:37.852+0200 [INFO]  CLI command args: []string{"plan"}s
2023-06-26T16:49:37.853+0200 [TRACE] Meta.Backend: built configuration for "s3" backend with hash value 3579742363
2023-06-26T16:49:37.853+0200 [TRACE] Preserving existing state lineage "REDACTED-f9cbb9069c3e"
2023-06-26T16:49:37.853+0200 [TRACE] Preserving existing state lineage "REDACTED-f9cbb9069c3e"
2023-06-26T16:49:37.854+0200 [TRACE] Meta.Backend: working directory was previously initialized for "s3" backend
2023-06-26T16:49:37.854+0200 [TRACE] Meta.Backend: using already-initialized, unchanged "s3" backend configuration
2023-06-26T16:49:37.855+0200 [INFO]  Attempting to use session-derived credentials
```

### Would you like to implement a fix?

None

**Comment by RafaelClovr** [comment]
https://github.com/hashicorp/terraform/issues/33421#event-9619222060 @justinretzolk  here is the closed Issue

**Comment by RafaelClovr** [comment]
I updaded aws cli to to the latest version:
`aws-cli/2.12.5 Python/3.11.4 Linux/6.3.8-200.fc38.x86_64 exe/x86_64.fedora.38 prompt/off` , still having the issue.

**Comment by RafaelClovr** [comment]
Provider` 5.6.1`

**Comment by gdavison** [comment]
Hi @RafaelClovr, this may be fixed in Terraform v1.6. Are you able to try with `1.6.0-beta2`?

**Comment by RafaelClovr** [comment]
No, I am not using any beta.

**Comment by quincy-codametrix** [comment]
This doesn't appear to be an issue with version 1.6.2

**Comment by gdavison** [comment]
This appears to be resolved in Terraform v1.6. If this does not work for you, please open a new issue
